src/scripts/import-open-api.ts

```diff
@@ -737,6 +740,21 @@ Path    : ${i.path}`),
   // tslint:enable:no-console
 };
 
+/**
+ * Get the url encoding function to be aliased at the module scope.
+ * This function is used to encode the path parameters.
+ *
+ * @param mode Either "uricomponent" or "rfc". "rfc" mode also encodes
```

Comment:
Do we really need this "rfc" mode? (you are the first one asking for this kind feature, so let's keep it simple)

Reply:
The thing is, our backend developer insists that we encode the parameters in accordance with RFC 3986. This means `!` and a bit more symbols have to be encoded, too. The biggest issue, however, is that I don't really see a clean, nice way to customize the encoding function. It all boils down to these options IMO:

- We can provide an `encodingFunction` option, and its value is a string, which is going to be interpolated into the resulting output file. This is the most effortless solution, however, this is very poor DX
- Same but the value of the option is a function. However we need to get the function's source code into the resulting file somehow, and we to get it typed. This seems not really feasible
- Provide an option `encodingFnAlias`, which when used in conjunction with `customImports` option will achieve the desired effect, i.e. you first declare a custom import with a custom encoding function in it, then specify to the code generator that you intend to use by using the `encodingFnAlias` option.
- Provide an option `encodingFn` with value of type `{ module: string; namedImport: string }`. This will ensure that a module containing a custom encoding function will get imported and used accordingly. However, this is a bit complicated.

So I decided to settle on two modes for now.

Reply:
Fair enough! Let's stick with this, but let's maybe call the option `rfc3986` to avoid any missunderstanding :wink: